tax/set.go

```diff
@@ -102,6 +102,10 @@ func combineExtKeys(cat *Category, rate *Rate) []cbc.Key {
 // prepare updates the Combo object's Percent and Retained properties using the base totals
 // as a source of additional data for making decisions.
 func (c *Combo) prepare(r *Regime, tags []cbc.Key, date cal.Date) error {
+	if err := r.CalculateObject(c); err != nil {
+		return err
+	}
```

Comment:
Did you test this at the bottom already? While not the case for GR, other countries may need some of the other calculations to be performed beforehand.

Reply:
I _think_ I saw that, for other structs, the regimes calculation are executed _before_ the GOBL ones. But I'll double-check.

Reply:
It might change according to the model... in this case there are preparations so it probably makes sense lower down.

Reply:
Agreed. I took care of it in [34df770](https://github.com/invopop/gobl/pull/307/commits/34df77054406dfea18c2e47471eff385cb992570)